analyzers/src/SonarAnalyzer.CSharp/Rules/MultipleVariableDeclarationCodeFix.cs

```diff
@@ -32,54 +27,38 @@ public class MultipleVariableDeclarationCodeFix : MultipleVariableDeclarationCod
     {
         protected override SyntaxNode CalculateNewRoot(SyntaxNode root, SyntaxNode node)
         {
-            if (!(node is VariableDeclaratorSyntax declarator))
+            if (node is not VariableDeclaratorSyntax { Parent: VariableDeclarationSyntax declaration })
             {
                 return root;
             }
 
-            if (!(declarator.Parent is VariableDeclarationSyntax declaration))
-            {
-                return root;
-            }
-
-            var newDeclarations = declaration.Variables.Select(variable =>
-                SyntaxFactory.VariableDeclaration(
-                    declaration.Type.WithoutTrailingTrivia(),
-                    SyntaxFactory.SeparatedList(new[] { variable.WithLeadingTrivia(GetLeadingTriviaFor(variable)) })));
+            var newDeclarations = declaration.Variables.Select(variable => CreateNewDeclaration(variable, declaration));
 
             IEnumerable<SyntaxNode> newNodes;
 
-            if (!(declaration.Parent is FieldDeclarationSyntax fieldDeclaration))
+            if (declaration.Parent is not FieldDeclarationSyntax fieldDeclaration)
             {
-                if (!(declaration.Parent is LocalDeclarationStatementSyntax localDeclaration))
+                if (declaration.Parent is not LocalDeclarationStatementSyntax localDeclaration)
                 {
                     return root;
                 }
 
-                newNodes = newDeclarations
-                    .Select(decl =>
-                        SyntaxFactory.LocalDeclarationStatement(
-                            localDeclaration.Modifiers,
-                            decl));
+                newNodes = newDeclarations.Select(decl => SyntaxFactory.LocalDeclarationStatement(localDeclaration.Modifiers, decl));
```

Comment:
On multiple places in this file
```suggestion
                newNodes = newDeclarations.Select(x => SyntaxFactory.LocalDeclarationStatement(localDeclaration.Modifiers, x));
```